### PR TITLE
Lms/comprehension prompt health report performance

### PR DESCRIPTION
**Pull request by anathomical**
## WHAT
Refactor the Prompt Health query to be pure SQL rather than doing Ruby-based post-processing on a simpler query
## WHY
The loops in the post-processing were causing performance issues, so we're moving all the logic in to SQL in order to improve performance
## HOW
Write a relatively complex aggregate SQL query to do all of the calculations that we care about in SQL

### Notion Card Links
https://www.notion.so/quill/Comprehension-Prompt-Health-Endpoint-Performance-cefc25b427494bc18a87084909acc794

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  Yes
Have you deployed to Staging? | Not yet - deploying now!
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Design Review: If applicable, have you compared the coded design to the mockups? | N/A

### CODE REVIEW

File: services/QuillLMS/app/queries/prompt_feedback_history.rb [hunk 1]
```diff
@@ -1,128 +1,43 @@
 class PromptFeedbackHistory
 
     def self.run(activity_id)
-        promptwise_postprocessing promptwise_sessions(activity_id)
+        #promptwise_postprocessing promptwise_sessions(activity_id)
```
Comment (happythenewsad) [inline services/QuillLMS/app/queries/prompt_feedback_history.rb]: Missed a line here

**Review by happythenewsad — CHANGES_REQUESTED** [review]
This looks promising - nice work. Just a trivial change request. Also, it would be a good idea to run EXPLAIN ANALYZE on this PR and compare with `develop`.

File: services/QuillLMS/app/queries/prompt_feedback_history.rb (hunk 1 above)
Comment (anathomical) [reply services/QuillLMS/app/queries/prompt_feedback_history.rb]: Good catch.  Fixed.

**Comment by anathomical** [comment]
@happythenewsad We could run `EXPLAIN ANALYZE`, but in this case I'm not sure what it would demonstrate.  NewRelic tracing suggests that the time-spent in the current endpoint is mostly in Ruby, not in SQL, so we wouldn't actually expect this query to be more performant than the one we've been using on its face.  We just expect the whole call to be more performant.

**Comment by happythenewsad** [comment]
> We just expect the whole call to be more performant.

Yes, we expect the whole call to be more performant. I inquired because I want to avoid the SQL call getting too expensive. That said, I'm fine with trying it out in prod.

**Review by happythenewsad — APPROVED** [review]

**Pull request merged**